pgrx-examples/spi_srf/src/lib.rs

```diff
@@ -95,9 +95,11 @@ fn filter_by_breed(
         let tup_table = client.select(query, None, Some(args))?;
 
         let filtered = tup_table
-            .map(|row| (row["dog_name"].value(), row["dog_age"].value(), row["dog_breed"].value()))
-            .collect::<Vec<_>>();
-        Ok(TableIterator::new(filtered))
+            .map(|row| {
+                Ok((row["dog_name"].value()?, row["dog_age"].value()?, row["dog_breed"].value()?))
```

Comment:
It's not terrible, but I'm sure you agree that returning a Result from a closure is awkward at best.  In this case we `.collect::<Result<Vec<_>>>()` so it's not too terrible.  

I prefer the left side of this diff, but like I said elsewhere, I'm not willing to die on this hill.

Reply:
yeah, the left side is a bit nicer, but that type signature... that's what I really have it in for. so yeah, if I didn't believe it would open up better possibilities in short order, I wouldn't do it.

Reply:
> that's what I really have it in for

Yeah, that's why I'm not really gonna fight on this one.  I hate just typing out all that mess.  But once the return signature has been typed out, it makes it a lot easier to just write some code and return some values.

Reply:
Many question marks is much-preferable to having excessively-verbose nested type signatures, yeah.